main/observing/observation_run.py

```diff
@@ -140,6 +152,7 @@ def everything_ok(self):
 
             else:
```

Comment:
don't forget to add a threadcheck under the "while current_time < sunset_time" loop (currently line 146)